src/mixins/createStateful.ts

```diff
@@ -228,7 +228,7 @@ const createStateful: StatefulFactory = compose<StatefulMixin<State>, StatefulOp
 				if (id && stateFrom) {
 					instance.own(instance.observeState(id, stateFrom));
 				}
-				else if (id || stateFrom) {
+				else if (stateFrom) {
 					throw new TypeError('Factory requires options "id" and "stateFrom" to be supplied together.');
```

Comment:
The message now doesn't match the condition...  It should say something like `When "stateFrom" is supplied, "id" is required."`